assets/javascripts/discourse/components/event-date.gjs

```diff
@@ -0,0 +1,93 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import I18n from "discourse-i18n";
+import guessDateFormat from "../lib/guess-best-date-format";
+
+export default class EventDate extends Component {
+    @service siteSettings;
+
+    <template>
+    {{#if this.shouldRender}}
+        <div class="header-topic-title-suffix-outlet event-date-container">
```

Comment:
The indentation looks off on this template, we only use 2 spaces for indentation normally.